code/frameworks/angular/src/client/decorators.ts

```diff
@@ -29,6 +30,26 @@ export const moduleMetadata =
     };
   };
 
+/**
+ * Decorator to set the config options which are available during the application bootstrap operation
+ */
+export const applicationConfig =
+  <TArgs = any>(
+    /**
+     * Set of config options available during the application bootstrap operation.
+     */
+    config: ApplicationConfig
+  ): DecoratorFunction<AngularRenderer, TArgs> =>
+  (storyFn) => {
+    const story = storyFn();
+    const storyConfig = story.applicationConfig;
+
+    return {
+      ...story,
+      applicationConfig: storyConfig ?? config,
+    };
+  };
```

Comment:
> If it is because objects aren't as straightforward to merge, should we at least merge the currently known providers array and overwrite the other potential properties, until we know what they are if any more are ever added?

That was more or less the reason. But you're right. It does not match the expectation. I will merge the providers!